pkgs/development/interpreters/dart/update.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq
+
+set -euo pipefail
+
+CHANNELS=(stable beta dev)
+declare -A PLATFORMS=(
+    [macos-x64]=x86_64-darwin
+    [linux-ia32]=i686-linux
+    [linux-x64]=x86_64-linux
+    [linux-arm]=armv7l-linux
+    [linux-arm64]=aarch64-linux
+)
+BASE="https://storage.googleapis.com/dart-archive/channels"
+
+TMP=sources.tmp.json
+echo '' > $TMP
```

Comment:
```suggestion
touch $TMP
```

Reply:
If `$TMP` exists, e.g. after a failed update attempt, this will result in appending to a partial result rather than starting with an empty result.

Reply:
I think a better solution would be to use `mktemp` instead.

```
TMP=$(mktemp --suffix .tmp.json)
```

This will generate a file with a unique name.